sa_sigaction doesn't build on FreeBSD

I'm trying to build https://github.com/ifreund/river/ but it fails due to borked translation of `struct sigaction`. Unlike DragonFly `siginfo_t` is not defined, so FreeBSD uses `__siginfo` but... it's also not defined.
```zig
$ cat test.zig
const std = @import("std");

pub fn main() anyerror!void {
    var foo = std.os.Sigaction{
        .sigaction = null,
        .mask = std.os.empty_sigset,
        .flags = 0,
    };
}

$ zig build-exe test.zig
/usr/local/lib/zig/std/os/bits/freebsd.zig:745:35: error: use of undeclared identifier '__siginfo'
        __sa_sigaction: fn (i32, *__siginfo, usize) callconv(.C) void,
                                  ^
/usr/local/lib/zig/std/os/bits/freebsd.zig:743:5: note: while checking this field
    __sigaction_u: extern union {
    ^
./test.zig:4:31: note: referenced here
    var foo = std.os.Sigaction{
                              ^
```
https://github.com/ziglang/zig/blob/cc3bceea3de67f24e6c17d4c04a34a2301e097f8/lib/std/os/bits/freebsd.zig#L740-L753
https://github.com/ziglang/zig/blob/cc3bceea3de67f24e6c17d4c04a34a2301e097f8/lib/std/os/bits/dragonfly.zig#L520-L527

--- a/lib/std/debug.zig
+++ b/lib/std/debug.zig
@@ -1696,6 +1696,7 @@ fn getDebugInfoAllocator() *mem.Allocator {
 pub const have_segfault_handling_support = switch (builtin.os.tag) {
     .linux, .netbsd => true,
     .windows => true,
+    .freebsd => @hasDecl(os, "ucontext_t"),
     else => false,
 };
 pub const enable_segfault_handler: bool = if (@hasDecl(root, "enable_segfault_handler"))
@@ -1757,6 +1758,7 @@ fn handleSegfaultLinux(sig: i32, info: *const os.siginfo_t, ctx_ptr: ?*const c_v
 
     const addr = switch (builtin.os.tag) {
         .linux => @ptrToInt(info.fields.sigfault.addr),
+        .freebsd => @ptrToInt(info.addr),
         .netbsd => @ptrToInt(info.info.reason.fault.addr),
         else => unreachable,
     };
@@ -1781,8 +1783,16 @@ fn handleSegfaultLinux(sig: i32, info: *const os.siginfo_t, ctx_ptr: ?*const c_v
         },
         .x86_64 => {
             const ctx = @ptrCast(*const os.ucontext_t, @alignCast(@alignOf(os.ucontext_t), ctx_ptr));
-            const ip = @intCast(usize, ctx.mcontext.gregs[os.REG_RIP]);
-            const bp = @intCast(usize, ctx.mcontext.gregs[os.REG_RBP]);
+            const ip = switch (builtin.os.tag) {
+                .linux, .netbsd => @intCast(usize, ctx.mcontext.gregs[os.REG_RIP]),
+                .freebsd => @intCast(usize, ctx.mcontext.rip),
+                else => unreachable,
+            };
+            const bp = switch (builtin.os.tag) {
+                .linux, .netbsd => @intCast(usize, ctx.mcontext.gregs[os.REG_RBP]),
+                .freebsd => @intCast(usize, ctx.mcontext.rbp),
+                else => unreachable,
+            };
             dumpStackTraceFromBase(bp, ip);
         },
         .arm => {

--- a/lib/std/os/bits/freebsd.zig
+++ b/lib/std/os/bits/freebsd.zig
@@ -4,6 +4,7 @@
 // The MIT license requires this copyright notice to be included in all copies
 // and substantial portions of the software.
 const std = @import("../../std.zig");
+const builtin = std.builtin;
 const maxInt = std.math.maxInt;
 
 // See https://svnweb.freebsd.org/base/head/sys/sys/_types.h?view=co
@@ -814,6 +815,53 @@ pub const sigset_t = extern struct {
 };
 
 pub const empty_sigset = sigset_t{ .__bits = [_]u32{0} ** _SIG_WORDS };
+
+pub usingnamespace switch (builtin.arch) {
+    .x86_64 => struct {
+        pub const ucontext_t = extern struct {
+            sigmask: sigset_t,
+            mcontext: mcontext_t,
+            link: ?*ucontext_t,
+            stack: stack_t,
+            flags: c_int,
+            __spare__: [4]c_int,
+        };
+
+        /// XXX x86_64 specific
+        pub const mcontext_t = extern struct {
+            onstack: u64,
+            rdi: u64,
+            rsi: u64,
+            rdx: u64,
+            rcx: u64,
+            r8: u64,
+            r9: u64,
+            rax: u64,
+            rbx: u64,
+            rbp: u64,
+            r10: u64,
+            r11: u64,
+            r12: u64,
+            r13: u64,
+            r14: u64,
+            r15: u64,
+            trapno: u32,
+            fs: u16,
+            gs: u16,
+            addr: u64,
+            flags: u32,
+            es: u16,
+            ds: u16,
+            err: u64,
+            rip: u64,
+            cs: u64,
+            rflags: u64,
+            rsp: u64,
+            ss: u64,
+        };
+    },
+    else => struct {},
+};
 
 pub const EPERM = 1; // Operation not permitted
 pub const ENOENT = 2; // No such file or directory

